pkg/util/serviceaccount.go

```diff
@@ -83,14 +83,34 @@ func WaitForServiceAccountSecretCreation(client kubeclient.Interface, asObj *cor
 			}
 			return false, fmt.Errorf("failed to retrieve service account(%s/%s) from cluster, err: %v", asObj.Namespace, asObj.Name, err)
 		}
-		clusterSecret, err = GetTargetSecret(client, serviceAccount.Secrets, corev1.SecretTypeServiceAccountToken, asObj.Namespace)
+		if len(serviceAccount.Secrets) == 0 {
```

Comment:
triggering off the length of this list is not really correct... this list can contain the names of non-token secrets

Reply:
Thanks for the reminder.

Reply:
any better idea for this? 

should we just create the secret all the time? i test it work with k8s 1.18,1.23,1.24

Reply:
Yes. If you are using a secret based token, you should create it, not use the auto-generated one